lib/command/src/Obelisk/Command/Project.hs

```diff
@@ -312,9 +313,22 @@ nixShellRunConfig root isPure command = do
       , [cs]
       ])
 
+-- | Escape using ANSI C-style quotes $''
+-- This does not work with all shells! Ideally, we would control exactly which shell is used,
+-- down to its sourced configuration, throughout the obelisk environment. At this time, this
+-- is not feasible.
 bashEscape :: String -> String
 bashEscape = BSU.toString . bytes . bash . BSU.fromString
 
+-- | Escape using bourne style shell escaping
+-- This is not as robust, but is necessary if we are passing to a shell we don't control.
```

Comment:
How is it less robust?

Reply:
Not every byte or control sequence can be escaped in that way. $'' can escape literally anything.